tests/integration_test.py

```diff
@@ -36,15 +37,40 @@
 from .test import Cleanup
 from docker.errors import APIError
 
+import pytest
+
 # FIXME: missing tests for
 # export; history; insert; port; push; tag; get; load; stats
-DEFAULT_BASE_URL = os.environ.get('DOCKER_HOST')
-EXEC_DRIVER_IS_NATIVE = True
-NOT_ON_HOST = os.environ.get('NOT_ON_HOST', False)
 
 warnings.simplefilter('error')
 compare_version = docker.utils.compare_version
 
+_exec_driver = []
+
+
+def exec_driver_is_native():
+    if not _exec_driver:
+        c = docker_client()
+        _exec_driver.append(c.info()['ExecutionDriver'])
```

Comment:
A hack caused by my misunderstanding of what features existed in Python 2 and 3. Updated to use `global`.